go/libraries/doltcore/doltdb/doltdb.go

```diff
@@ -409,6 +411,70 @@ func (ddb *DoltDB) CommitWithParentSpecs(ctx context.Context, valHash hash.Hash,
 	return ddb.CommitWithParentCommits(ctx, valHash, dref, parentCommits, cm)
 }
 
+func (ddb *DoltDB) WriteCommitDanglingCommit(ctx context.Context, valHash hash.Hash, parentCommits []*Commit, cm *CommitMeta) (*Commit, error) {
```

Comment:
WriteDanglingCommit?

Reply:
Yeah it uses Noms terminology.  It means to write a commits data to the database without pointing a Noms dataset at the new Commit struct.  That's what  `commitSt, err = ddb.db.CommitDangling(ctx, val, commitOpts)` is doing. A Dolt branch is equivalent to a dataset in Noms.